app/submitter/convert_payload_0_0_1.py

```diff
@@ -127,8 +127,9 @@ def _get_checkbox_answer_data(
             if "detail_answer" in option:
                 detail_answer = answer_store.get_answer(option["detail_answer"]["id"])
                 # if the user has selected an option with a detail answer we need to find the detail answer value it refers to.
-                # the detail answer value can be empty, in this case we just use the main value (e.g. other)
-                user_answer = detail_answer.value or user_answer  # type: ignore
+                # the detail answer maybe None and the value could be empty, in this case we just use the main value (e.g. other)
+                if detail_answer:
+                    user_answer = detail_answer.value or user_answer  # type: ignore
```

Comment:
The `or` part isn't needed?

Reply:
I added the or as there is an existing test which expect the `user_answer` to be `Other` rather than empty when detail_answer.value doesn't exist

Reply:
The existing test is `test_converter_checkboxes_with_q_codes_and_empty_other_value`

Reply:
We don't store empty answers?

Reply:
I've updated this now but can't get past the mypy type hinting error  🤯

Reply:
Removed invalid test that was causing confusion